crypto/signature.go

```diff
@@ -0,0 +1,85 @@
+package neofscrypto
+
+import (
+	"fmt"
+
+	"github.com/nspcc-dev/neofs-api-go/v2/refs"
+)
+
+// Signature represents a confirmation of data integrity received by the
+// digital signature mechanism.
+//
+// Signature is mutually compatible with github.com/nspcc-dev/neofs-api-go/v2/refs.Signature
+// message. See ReadFromV2 / WriteToV2 methods.
+//
+// Note that direct typecast is not safe and may result in loss of compatibility:
+// 	_ = Signature(refs.Signature{}) // not recommended
+type Signature refs.Signature
+
+// ReadFromV2 reads Signature from the refs.Signature message.
+//
+// See also WriteToV2.
+func (x *Signature) ReadFromV2(m refs.Signature) {
+	*x = Signature(m)
+}
+
+// WriteToV2 writes Signature to the refs.Signature message.
+// The message must not be nil.
+//
+// See also ReadFromV2.
+func (x Signature) WriteToV2(m *refs.Signature) {
+	*m = (refs.Signature)(x)
+}
+
+// Calculate signs data using Signer and encodes public key for subsequent
+// verification.
+//
+// Signer MUST NOT be nil.
+//
+// See also Verify.
+func (x *Signature) Calculate(signer Signer, data []byte) error {
+	signature, err := signer.Sign(data)
+	if err != nil {
+		return fmt.Errorf("signer %T failure: %w", signer, err)
+	}
+
+	key := make([]byte, signer.MaxPublicKeyEncodedSize())
+	key = key[:signer.EncodePublicKey(key)]
+
+	m := (*refs.Signature)(x)
+
+	scheme := refs.SignatureScheme(signer.Scheme())
+	if scheme > 0 {
+		scheme-- // to sync numeric values
```

Comment:
why not `switch`?

Reply:
In order to not loose custom/unsupported values.

Reply:
and what conversion rule is expected when that code faces an unknown scheme (lets say... `5`)? did not really get the idea

Reply:
I synced values after https://github.com/nspcc-dev/neofs-sdk-go/pull/190#discussion_r852091743.